CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatChannel.lua

```diff
@@ -0,0 +1,155 @@
+local source = [[
+local module = {}
+--////////////////////////////// Include
+--//////////////////////////////////////
+local modulesFolder = script.Parent
+local moduleTransparencyTweener = require(modulesFolder:WaitForChild("TransparencyTweener"))
+local ClassMaker = require(modulesFolder:WaitForChild("ClassMaker"))
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+local methods = {}
+
+local function CreateGuiObject()
+	local BaseFrame = Instance.new("Frame")
+	BaseFrame.Size = UDim2.new(1, 0, 1, 0)
+	BaseFrame.BackgroundTransparency = 1
+
+	local Scroller = Instance.new("ScrollingFrame", BaseFrame)
+	Scroller.Name = "Scroller"
+	Scroller.BackgroundTransparency = 1
+	Scroller.BorderSizePixel = 0
+	Scroller.Position = UDim2.new(0, 0, 0, 3)
+	Scroller.Size = UDim2.new(1, -4, 1, -6)
+	Scroller.CanvasSize = UDim2.new(0, 0, 0, 0)
+	Scroller.ScrollBarThickness = module.ScrollBarThickness
+	Scroller.Active = false
+
+	return BaseFrame, Scroller
+end
+
+function methods:Destroy()
+	self.GuiObject:Destroy()
+end
+
+function methods:SetActive(active)
+	self.GuiObject.Visible = active
+end
+
+function methods:AddMessageLabelToLog(messageObject)
+	self.TextTweener:RegisterTweenObjectProperty(messageObject.Tweener, "Transparency")
+
+	table.insert(self.MessageObjectLog, messageObject)
+	self:PositionMessageLabelInWindow(messageObject)
+
+	if (#self.MessageObjectLog > 50) then
+		self:RemoveLastMessageLabelFromLog()
+	end
+end
+
+function methods:RemoveLastMessageLabelFromLog()
+	local lastMessage = self.MessageObjectLog[1]
+	local posOffset = UDim2.new(0, 0, 0, lastMessage.BaseFrame.AbsoluteSize.Y)
+
+	lastMessage:Destroy()
+	table.remove(self.MessageObjectLog, 1)
+
+	for i, messageObject in pairs(self.MessageObjectLog) do
+		messageObject.BaseFrame.Position = messageObject.BaseFrame.Position - posOffset
+	end
+
+	self.Scroller.CanvasSize = self.Scroller.CanvasSize - posOffset
+end
+
+function methods:PositionMessageLabelInWindow(messageObject)
+	local baseFrame = messageObject.BaseFrame
+	local baseMessage = messageObject.BaseMessage
+
+	baseFrame.Parent = self.Scroller
+	baseFrame.Position = UDim2.new(0, 0, 0, self.Scroller.CanvasSize.Y.Offset)
+	
+	--// This looks stupid, but it's actually necessary.
+	--// TextBounds wont be calculated correctly unless it has enough space.
+	baseFrame.Size = UDim2.new(1, 0, 0, 1000)
```

Comment:
I don't think this particular situation is a big deal. It's only one extra line rather than calling a function from another module to do the calculation. Plus, the comments make it clear what is happening.